pa_matchbatch.sh

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
+set -e
+unset CDPATH
+
+# where to find the agent; can be overridden
+puppet_agent_repo="git@github.com:puppetlabs/puppet-agent.git"
+PUPPET_AGENT_URL=${PUPPET_AGENT_URL:-${puppet_agent_repo}}
+
+# where to find the ticketmatch.rb script; can be overridden
+TICKETMATCH_PATH=${TICKETMATCH_PATH:-$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )}
+
+# shorthand
+pushd() {
+	builtin pushd "$@" > /dev/null
+}
+
+popd() {
+	builtin popd "$@" > /dev/null
+}
+
+# get rev hashes in the form [to_rev]|[url]
+getComponentRevMap() {
+	# looking for components not pinned to a 'refs/tags' element, and of those, filtering out (keeping) the ones owned by puppetlabs
+	for componentName in $(for componentFile in $(grep -lv refs/tags configs/components/*.json); do grep -l puppetlabs ${componentFile}; done); do 
+		ruby -rjson -e'j = JSON.parse(STDIN.read); printf(" %s|%s", j["ref"], j["url"])' < ${componentName}
+	done
+}
+
+# TODO: more robust version matches?
+getFixVerFor() {
+	case ${1} in
+		puppet-resource_api)
+			git log -1 --no-merges --oneline --grep='Release prep' | sed -e "s/.*prep for v\(.*$\)/\1/"
+		;;
+		*)
+			git log -1 --no-merges --oneline -E --grep='\(packaging\) .* version' | sed -e "s/.*'\(.*\)'.*/\1/"
```

Comment:
So there is a really stupid subtlety here and that's merge-ups. With merge-ups, we can have a version bump from a previous maint branch "override" (in terms of chronological order) the version bump in our current branch/ref. I found this out after a test run of your script.

It might be better to just read the version from the repo directly. There's some code you can copy-paste here for that: https://github.com/puppetlabs/ci-job-configs/blob/master/resources/scripts/vanagon-helpers.sh#L372-L448

^ The "{{" brace stuff is b/c of interpolation; all of the "{{" or "}}" you see should be replaced with "{" or "}" respectively if you want to use it in your script. (You can probably ignore the other cases in the read_repo_version function b/c those were added by different teams).